subscription-descriptor-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/SubscriptionDescriptorFileParserTest.java

```diff
@@ -14,38 +14,50 @@
 import uk.gov.justice.domain.subscriptiondescriptor.Subscription;
 import uk.gov.justice.domain.subscriptiondescriptor.SubscriptionDescriptor;
 
-import java.io.FileNotFoundException;
-import java.net.URISyntaxException;
+import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.util.Collection;
 import java.util.List;
 
-import com.fasterxml.jackson.databind.JsonMappingException;
+import org.everit.json.schema.ValidationException;
 import org.junit.Test;
 
 public class SubscriptionDescriptorFileParserTest {
 
     final Path baseDir = get("src/test/resources/");
-    final SubscriptionDescriptorFileParser parser = new SubscriptionDescriptorFileParser();
 
+    @Test
+    public void shouldThrowFileNotFoundException() {
+        final SubscriptionDescriptorFileParser subscriptionDescriptorFileParser = new SubscriptionDescriptorFileParserFactory().create();
+        final List<Path> paths = asList(get("no-subscription.yaml"));
+        try {
+            subscriptionDescriptorFileParser.parse(baseDir, paths);
+            fail();
+        } catch (RuntimeException re) {
```

Comment:
The test should be catching the expected exception and asserting that it is the exception thrown

Reply:
Fixed